### DISCUSSION TITLE
No way to obtain a delta token for whole drive with go

### DISCUSSION THREAD

**Discussion by luigi-fast** [microsoftgraph/msgraph-sdk-go#543] [Q&A] [answered]
Hello,
I would want to obtain a delta token for a whole drive with golang but the documentation indicate only how to proceed with a driveItem:

Java example:
graphClient.me().drive().root().delta...

but the same example for Go:
graphClient.Drives().ByDriveId("drive-id").Items().ByItemId("driveItem-id").Delta...

Do you know how I can obtain a delta for an entire drive with go?

**Comment by baywet** [comment, marked as answer]
For reference [here is the documentation](https://learn.microsoft.com/en-us/graph/api/driveitem-delta?view=graph-rest-1.0&tabs=http)

Have you tried ?

```Go
graphClient.Drives().ByDriveId("drive-id").Items().ByItemId("root").Delta...
```

What's not clear from the documentation to me is whether this will enumerate changes from the direct children of the root folder or also include subfolders. But it's worth giving a try.

**Reply by luigi-fast** [reply]
Thanks for the suggestion! I appreciate your help. I'll give it a try with ByItemId("root") to obtain the root item and see how it works for enumerating changes. 

It can be a bit confusing when the Go implementation has a method like Drives().ByDriveId({drive-id}).Root(), when the Java code have a corresponding delta method for the object returned by root method.  I expected the go implementation to have the same method.

thanks for your quick reply

**Reply by baywet** [reply]
There's a bit of historical reasons for that:
1. Root is not a thing in the metadata we use to generate the code.
2. The java SDK is based on an older generator (working to migrate to kiota, go is using kiota), and that older generator has a "fix" specifically for that.
